vmdb/db/fixtures/ae_datastore/ManageIQ/Cloud/Orchestration/Retirement/Email.class/__methods__/stack_retirement_emails.rb

```diff
@@ -0,0 +1,121 @@
+#
+# Description: This method sends out retirement emails when the following events are raised:
+#
+# Events: stack_retire_warn, stack_retired, stack_entered_retirement
+#
+# Model Notes:
+# 1. to_email_address - used to specify an email address in the case where the
+#    vm's owner does not have an  email address. To specify more than one email
+#    address separate email address with commas. (I.e. admin@company.com,user@company.com)
+# 2. from_email_address - used to specify an email address in the event the
+#    requester replies to the email
+# 3. signature - used to stamp the email with a custom signature
+#
+stack = $evm.object['orchestration_stack']
+if stack.nil?
+  stack_id = $evm.object['orchestration_stack_id'].to_i
+  stack = $evm.vmdb('orchestration_stack', stack_id) unless stack_id == 0
+end
+
+if stack.nil?
+  stack = $evm.root['orchestration_stack']
+  if stack.nil?
+    stack_id = $evm.root['orchestration_stack_id'].to_i
+    stack = $evm.vmdb('orchestration_stack', stack_id) unless stack_id == 0
+  end
+end
+
+prov = $evm.root['miq_provision_request'] || $evm.root['miq_provision']
+# if stack.nil?
+#   stack = prov.orchestration_stack unless prov.nil?
+# end
+
+stack = prov.orchestration_stack if prov && stack.nil?
```

Comment:
Stacks are provisioned through services so I do not see how you would ever have a miq_provision_request or miq_provision key here.  Should this reference `service_template_provision_request` and 'service_template_provision_task`?  Also the commented code should be removed.